### PR TITLE
Update XLMR quality and restructure file structure

### PR DESCRIPTION
**Pull request by erichan1**
Major update to XLMR, with the following components
1. Restructuring of the XLMR benchmark into xlmr.py (main benchmark), xlmr_data.py (synthetic data generation), xlmr_utils.py (util functions), xlmr_parser.py (command line args parser). 
2. Creation of the sample_sequence_length(), which allows one to sample from a distribution of sequence lengths. This means the synthetic input data can be more closely matched to a real world use-case's data distribution. 
3. Switching from xlmr.model(input) to xlmr.extract_features(input). The difference: the former outputs tensors of shape (batch_size, seq_len, vocab_size) and the latter ouputs shape of (batch_size, seq_len, embedding_size). Typically vocab size is huge, like 250k, while embedding size is 1k, which means new output is much smaller. This is also more accurate to real use cases - typically we care about getting the embeddings from the last encoder layer like in the latter case, not getting the probability (really logits, but same diff) of every possible token in the vocabulary being in the input sequences.
4. Fixing of a memory leak that caused GPU OOMs by calling del on the output tensors from the model's forward and backwards pass. See more in this [comment](https://github.com/facebookresearch/FAMBench/pull/70/files?authenticity_token=rtFV0c9e35WhYv1goGHviMnh3Ow7abuIcO8hO8qHyX0MVi9zLIAEBn9HM8yA67oNW3JstTuTyJhvp6faupsPSw%3D%3D&file-filters%5B%5D=.py#r769141498) on this PR.

### CODE REVIEW

File: benchmarks/xlmr/ootb/xlmr.py [hunk 1]
```diff
@@ -76,40 +67,54 @@ def train(model, x_l, y_true_l, device=None, logger=None):
...
-        y_pred = model(x)
-        y_true = y_true.long()
-        loss = F.cross_entropy(y_pred[0], y_true[:,0,:]) # TODO: fix y_true data input hack
+        y_pred = xlmr.extract_features(x)
+        loss = F.cross_entropy(y_pred, y_true) 
         loss.backward()
         optimizer.step()
         optimizer.zero_grad() 
+
+        del y_pred
```
Comment (erichan1) [inline benchmarks/xlmr/ootb/xlmr.py:82]: This marks y_pred and loss as "deletable". Thus, the python garbage collector can go recover the memory, and the memory used during the computation of y_pred on gpu can also be recovered. If these aren't included then we see increasing memory usage as batches are run through the model, which shouldn't happen. Same thing in the inference function above.

Comment (orionr) [reply benchmarks/xlmr/ootb/xlmr.py:82]: Do you want to add a comment to the code about this? Seems non-obvious

File: benchmarks/xlmr/ootb/xlmr_utils.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
+
+import numpy as np
+
+"""
+Util functions for xlmr benchmark
```
Comment (orionr) [inline benchmarks/xlmr/ootb/xlmr_utils.py:5]: Nit: XLMR or XLM-R?

**Review by orionr — APPROVED** [review]
Minor nits! LGTM

File: benchmarks/xlmr/ootb/xlmr.py (hunk 1 above)
Comment (erichan1) [reply benchmarks/xlmr/ootb/xlmr.py:82]: Makes sense - I'll add a comment there. I was thinking if someone really needs to know they can look at git blame, but people may copy this code or just not look at git.

File: benchmarks/xlmr/ootb/xlmr_utils.py (hunk 2 above)
Comment (erichan1) [reply benchmarks/xlmr/ootb/xlmr_utils.py:5]: Good call. I use these pretty interchangeably. XLM-R is more accurate so I'll keep to that in comments and xlmr in code.
